### PR TITLE
fixes: object schema carry properties added by raw to $ref

### LINKED ISSUE #216: Object schema does not carry properties added by .raw() to a $ref
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.10.2

### Plugin version

_No response_

### Node.js version

16.14.0

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

5.10.167-1-MANJARO

### Description

I think there is a problem with using ref + raw in the object schema in that the values added by the `raw()` function are removed or not included

### Steps to Reproduce

correct
```javascript
 fluent.ref('foo').raw({test: true}).valueOf()
{ '$ref': 'foo', test: true }
```

incorrect

```javascript
fluent.object().prop('a', fluent.string()).prop('test', fluent.ref('foo').raw({test: true})).valueOf()
{
  '$schema': 'http://json-schema.org/draft-07/schema#',
  type: 'object',
  properties: { a: { type: 'string' }, test: { '$ref': 'foo' } }
}
```

I suspect it is from the object schemas handling of ref

https://github.com/fastify/fluent-json-schema/blob/ab242d2a88973c78365a042fc2e9568208a28872/src/ObjectSchema.js#L321-L331

### Expected Behavior

I'd expect the properties defiend by `.raw()` to be carried unconditionally to the object they are defined

### PR DESCRIPTION
**Pull request by dancastillo**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: src/ObjectSchema.js [hunk 1]
```diff
@@ -323,7 +324,9 @@ const ObjectSchema = ({ schema = initialState, ...options } = {}) => {
           ...schema,
           [target]: [
             ...schema[target],
-            $ref ? { name, $ref } : Object.assign({}, { name }, attributes)
+            $ref
+              ? (Object.keys(restAttributes).length ? { name, $ref, ...restAttributes } : { name, $ref })
+              : Object.assign({}, { name }, attributes)
```
Comment (Uzlopak) [inline src/ObjectSchema.js:329]: I guess we dont need the first empty object if we create the second one on the fly too?

```suggestion
              : Object.assign({ name }, attributes)
```

File: src/ObjectSchema.js [hunk 2]
```diff
@@ -323,7 +324,9 @@ const ObjectSchema = ({ schema = initialState, ...options } = {}) => {
           ...schema,
           [target]: [
             ...schema[target],
-            $ref ? { name, $ref } : Object.assign({}, { name }, attributes)
+            $ref
+              ? (Object.keys(restAttributes).length ? { name, $ref, ...restAttributes } : { name, $ref })
```
Comment (Uzlopak) [inline src/ObjectSchema.js:328]: Why not

```suggestion
              ? { name, $ref, ...restAttributes }
```

Comment (Uzlopak) [reply src/ObjectSchema.js:328]: The distinction makes no difference imho. If the object has no keys then spreading them will not add any key.

**Comment by Uzlopak** [comment]
@Fdawgs 
Any comment from you regarding Object.assign vs. Spread operator? Here we mix both techniques.

File: src/ObjectSchema.js (hunk 2 above)
Comment (dancastillo) [reply src/ObjectSchema.js]: Good catch updated to suggested change

**Comment by Fdawgs** [comment]
> @Fdawgs Any comment from you regarding Object.assign vs. Spread operator? Here we mix both techniques.

Best to just go all in on the spready boy?

**Review by Uzlopak — APPROVED** [review]
Lgtm

**Review by Fdawgs — APPROVED** [review]

**Comment by aboutlo** [comment]
@dancastillo thank you for your contribution! 💪

**Pull request merged**